File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/WritePropertyListValuesResponse.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.StatusCode;
import org.opcfoundation.webserver.types.common.UaChildId;

import java.util.HashMap;
import java.util.Map;

public class WritePropertyListValuesResponse {
    private final Map<UaChildId, StatusCode> results;

    public WritePropertyListValuesResponse()
    {
        results = new HashMap<>();
    }

    public void setWriteValueResult(String propertyId, StatusCode code)
    {
        results.put(new UaChildId(propertyId), code);
    }

    public void setWriteValueResult(UaChildId subPropertyId,StatusCode code)
    {
        results.put(subPropertyId, code);
    }

    public Map<UaChildId, StatusCode> getResults() {
        return results;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/WritePropertyListValuesResponse.ts
import { UaStatusCode } from "opcua-webapi-ts";
import { UaChildId } from "../../../types/common/UaChildId";

export class WritePropertyListValuesResponse {
    private readonly _results: Map<string, UaStatusCode>;

    constructor() {
        this._results = new Map();
    }

    setWriteValueResult(propertyId: string | UaChildId, code: UaStatusCode): void {
        this._results.set(this.toKey(propertyId), code);
    }

    get results(): Map<string, UaStatusCode> {
        return this._results;
    }

    private toKey(propertyId: string | UaChildId): string {
        if (typeof propertyId === "string") {
            return propertyId;
        }

        return propertyId.toString();
    }
}